src/main/kotlin/io/hsar/wh40k/combatsimulator/model/World.kt

```diff
@@ -64,6 +49,24 @@ data class World(val friendlyForces: MutableList<UnitInstance>, val enemyForces:
     fun getPosition(unit: UnitInstance): MapPosition {
         return this.unitPositions.getValue(unit)
     }
+
+    fun findDead(): List<UnitInstance> {
+        return listOf(findDeadInternal(friendlyForces),findDeadInternal(enemyForces)).flatten()
+    }
+
+    private fun findDeadInternal(unitList: MutableList<UnitInstance>): List<UnitInstance> {
+        return unitList.filter { unitInstance ->
+            when(val health = unitInstance.currentAttributes[Attribute.CURRENT_HEALTH]) {
+                is NumericValue -> health.value <= 0  // simplification
+                else -> throw RuntimeException("Current health must be a NumericValue")
+            }
+        }
+    }
+
+    fun removeUnits(unitsToRemove: List<UnitInstance>) {
```

Comment:
[Minor] A very small note that you can safely use the more abstract interface `Collection<>` here, not that it matters is 90% of cases.